synapse/handlers/account_validity.py

```diff
@@ -27,6 +29,14 @@
 
 logger = logging.getLogger(__name__)
 
+IS_USER_EXPIRED_CALLBACK = Callable[[str], Awaitable[Optional[bool]]]
+ON_USER_REGISTRATION_CALLBACK = Callable[[str], Awaitable]
+# Legacy callbacks to allow for a smoother transition between the legacy native account
+# validity feature and synapse-email-account-validity.
```

Comment:
As I said on our call: I think it would be good to expand this a bit to explain the status of these "legacy" callbacks, why they are needed, and how/when we hope to get rid of them.

Reply:
I assume this has already been fixed by https://github.com/matrix-org/synapse/pull/9884/commits/f8754fed56e93de206c561877b6fdfc60fa9191c